addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/handler/SolarEdgeAggregateDataPolling.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.solaredge.handler;
+
+import org.openhab.binding.solaredge.internal.command.AggregateDataUpdate;
+import org.openhab.binding.solaredge.internal.command.AggregatePeriod;
+import org.openhab.binding.solaredge.internal.command.SolarEdgeCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Polling worker class. This is responsible for periodic polling of sensor data.
+ *
+ * @author Alexander Friese - initial contribution
+ */
+public class SolarEdgeAggregateDataPolling implements Runnable {
+    /**
+     * Logger
+     */
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    /**
+     * Handler for delegation to callbacks.
+     */
+    private final SolarEdgeHandler handler;
+
+    /**
+     * Constructor.
+     *
+     * @param handler handler which handles results of polling
+     */
+    public SolarEdgeAggregateDataPolling(SolarEdgeHandler handler) {
+        this.handler = handler;
+    }
+
+    /**
+     * Poll the SolarEdge Webservice one time per call.
+     */
+    @Override
+    public void run() {
+        if (handler.getWebInterface() != null) {
+            logger.debug("polling SolarEdge aggregate data {}", handler.getConfiguration());
+
+            SolarEdgeCommand adu_day = new AggregateDataUpdate(handler, AggregatePeriod.DAY);
+            SolarEdgeCommand adu_week = new AggregateDataUpdate(handler, AggregatePeriod.WEEK);
+            SolarEdgeCommand adu_month = new AggregateDataUpdate(handler, AggregatePeriod.MONTH);
+            SolarEdgeCommand adu_year = new AggregateDataUpdate(handler, AggregatePeriod.YEAR);
+
+            try {
+                handler.getWebInterface().executeCommand(adu_day);
+                handler.getWebInterface().executeCommand(adu_week);
+                handler.getWebInterface().executeCommand(adu_month);
+                handler.getWebInterface().executeCommand(adu_year);
+            } catch (RuntimeException e) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Caught Exception: {}", e.getMessage(), e);
```

Comment:
Please add some real context, what are we doing, what went wrong?

Reply:
As I described before this is primary to get noticed about any exception that occours. I had a few NPEs in the past because the data returned by the API was suddenly different and values were null.

The exception was however hidden and the binding just stopped working silently so I added those try catch blocks at the binding's boundries.